content/blog/why-is-redfish®-different-from-other-rest-apis-part-1.md

```diff
@@ -0,0 +1,134 @@
+
+![Figure 6: Universal backplane model](/img/fig6-storagechassis.png "Figure 6: Universal backplane model")
+
+### Naming convention of collection members
+
+The naming of collection members is specific to each implementation of the Redfish® service. The BMC endpoint for an HPE server of type ProLiant is: `/redfish/v1/Managers/1`. That of an OpenBMC is: `/redfish/v1/bmc` and that of a Superdome is `/redfish/v1/Managers/RMC`. Dell, SuperMicro, Lenovo probably have their own way of naming their BMCs, while remaining compliant with the specifications.
+
+The naming flexibility given to Redfish® services generates extra work for client developers: their code has to discover the names of collection members before being able to access their properties. If they assume a specific URI, their code will fail when used against another computer maker or after a firmware update.
+
+## Registries
+
+The data modeling described above don't represent resources that have interdependencies (i.e., BIOS attribute %1 depends of attribute %2) as well as information or error messages and their arguments (e.g.: "error on disk %s in location %s"). To address this problem, Redfish® uses "registries".
+
+Figure 7 is extracted from a BIOS registry which has the effect of prohibiting the activation of Windows secure mode support if the server does not contain a TPM (Trusted Platform Module). It is interesting to note that the BIOS registry also describes the GUI menus (`MapToProperty=GrayOut`).
+
+![Figure 7: Example of interdependency in BIOS registry](/img/fig7-registrebios.png "Figure 7: Example of interdependency in BIOS registry")
+
+Figure 8 is taken from the base message registry. It shows the modeling of the “Access prohibited” message with the name of the prohibited resource as an argument (%1).
+
+![Figure 8: Modeling an error message with an argument](/img/fig8-registredesmessagesdebase.png "Figure 8: Modeling an error message with an argument")
+
+## Self-sufficient model
+
+Redfish® is considered "self-sufficient" because all (or almost) of the information regarding data modeling, operations and possible actions is documented and programmatically accessible. Client programmers no longer have to consult paper specifications before implementing them. They can transfer this task to their code in order to ensure a certain portability in time and space.
+
+### Accessing Schemas and registries
+
+Redfish® provides access to schemas and registries through the following endpoints:
+
+* `/redfish/v1/JsonSchemas`
+* `/redfish/v1/Registries`
+
+These URIs contain links to all schemas and registries used by the service. They point to documents stored in the BMC, if it has the storage capacity, or to the official DMTF website. In the later case, Redfish® clients must have a mean to access the Internet to download those documents. They are be helpful to identify property interdependencies, supported values or read-write capability.
```

Comment:
In the third sentence, I think you mean "In the latter case,..." And you might mean "must have a means to access the Internet"

In the final sentence, it should probably read:
"They are helpful to identify..." (don't need the word "be")